src/components/pill-selector-dropdown/PillSelectorDropdown.js

```diff
@@ -104,15 +110,28 @@ class PillSelectorDropdown extends React.Component<Props, State> {
             pills = pills.filter(pill => validator(pill));
         }
 
-        return pills.map(pill => ({
-            displayText: pill,
-            text: pill, // deprecated, left for backwards compatibility
-            value: pill,
-        }));
+        const normalizedPills = pills.map(pill =>
```

Comment:
Note that this puts the onus on the caller to also write a validator that handles the custom values returned by their parser